### PR TITLE
COMMON: Check CRC16 in Stuffit

### PR DESCRIPTION
**Pull request by phcoder**
This improves file corruption checking but due to its breakage potential I propose to shunt it to the next version. Opening PR so it doesn't get lost

### CODE REVIEW

**Comment by bluegr** [comment]
Would it make sense to include this, but not make it fatal?
i.e. change the errors to warnings

**Comment by lotharsm** [comment]
I think we could drop the "DONTMERGE" tag here. Branching of 2.7.x already happened, and we don't have to backport those changes to the 2.7.x branch.

**Comment by bluegr** [comment]
> I think we could drop the "DONTMERGE" tag here. Branching of 2.7.x already happened, and we don't have to backport those changes to the 2.7.x branch.

Indeed. Done

File: common/compression/stuffit.cpp [hunk 1]
```diff
@@ -140,37 +142,47 @@ bool StuffItArchive::open(Common::SeekableReadStream *stream) {
...
+		uint32 dataForkCompressedSize = headStream.readUint32BE();
+		uint16 resForkCRC = headStream.readUint16BE();
+		uint16 dataForkCRC = headStream.readUint16BE();
+		headStream.skip(6); // unknown
+		uint16 headerCRC = headStream.readUint16BE();
+
+		uint16 actualHeaderCRC = crc.crcFast(header, sizeof(header) - 2);
+
+		if (actualHeaderCRC != headerCRC)
+			error ("Header CRC mismatch: %04x vs %04x", actualHeaderCRC, headerCRC);
```
Comment (sev-) [inline common/compression/stuffit.cpp:185]: May it be possible to add here `StuffItArchive::open(): ` prefix and also the name of the broken file? Same for the `error()` below

Comment (bluegr) [inline common/compression/stuffit.cpp:185]:
```suggestion
			error ("StuffItArchive::open(): Header CRC mismatch: %04x vs %04x", actualHeaderCRC, headerCRC);
```

File: common/compression/stuffit.cpp [hunk 2]
```diff
@@ -274,6 +288,12 @@ Common::SharedArchiveContents StuffItArchive::readContentsForPath(const Common::
 		return Common::SharedArchiveContents();
 	}
 
+	uint16 actualCRC = Common::CRC16().crcFast(uncompressedBlock, entry.uncompressedSize);
+
+	if (actualCRC != entry.crc) {
+		error("CRC mismatch: %04x vs %04x", actualCRC, entry.crc);
```
Comment (bluegr) [inline common/compression/stuffit.cpp:294]:
```suggestion
		error("StuffItArchive::readContentsForPath(): CRC mismatch: %04x vs %04x for file %s", actualCRC, entry.crc, name.c_str());
```

**Comment by bluegr** [comment]
The suggestions to improve the error messages have been applied.
This looks good to be merged now. Thanks!

**Pull request merged**